src/components/App/OptionDropdown.vue

```diff
@@ -0,0 +1,199 @@
+
+export default {
+  name: "OptionDropdown",
+  data() {
+    return {
+      showDropdown: false, // whether to show the dropdown for choosing an option
+      defaultOptionMarginRem: 2, // default margin-top (in rem) of the option dropdown without any scrolling effect
+      optionBoxStyling: {}, // styling for the option dropdown
+    };
+  },
+  props: {
+    options: {
+      // the list of options in the dropdown
+      type: Array,
+      default: () => [],
+    },
+    scrollY: {
+        // the number of pixels have been scrolled vertically
+        default: 0,
+        type: Number
+    },
+    isTouchDevice: {
+        // whether the device being used on is a touch device
+        default: false,
+        type: Boolean
+    }
+  },
+  computed: {
+    defaultOptionMargin() {
+        // default margin-top (in pixels) of the option dropdown without any scrolling effect
+        return this.convertRemToPixels(this.defaultOptionMarginRem)
+    },
+    dropdownIconClass() {
+        // class for the dropdown icon
+      return [
+        { "transform rotate-180": this.showDropdown },
+        "transition ease duration-800 text-gray-600",
+      ];
+    },
+  },
+  watch: {
+    defaultOptionMarginRem() {
+        // update option box margin when the default top margin is updated
+        this.setOptionBoxStyling()
+    },
+    scrollY() {
+        // update option box margin when the page is scrolled is updated
+        this.setOptionBoxStyling()
+    },
+    showDropdown() {
+        // if the dropdown is closed, reset the option box's margin-top
+        if (!this.showDropdown) {
+            this.defaultOptionMarginRem = 2
+        }
+    }
+  },
+  created() {
+      // set the option box margin when the page is created
+      this.setOptionBoxStyling()
+  },
+  methods: {
+    ...Utilties,
+    optionClass(index) {
+        // class for each option
+        return {
+            'opacity-50 cursor-not-allowed': this.isOptionDisabled(index),
+            'hover:bg-primary hover:cursor-pointer hover:text-white': !this.isOptionDisabled(index)
+        }
+    },
+    isOptionDisabled(index) {
+        // whether the option at the given index is disabled
+        return this.options[index].disabled
+    },
+    setOptionBoxStyling() {
+        // sets the option box margin based on various parameters
+        if (!this.scrollY) {
+            this.optionBoxStyling = { 'margin-top': `${this.defaultOptionMarginRem}rem` }
+        } else {
+            this.optionBoxStyling = { 'margin-top': `-${this.scrollY - this.defaultOptionMargin}px`}
+        }
+    },
+    convertRemToPixels(rem) {
+        // converts value in rem to pixels
+        return rem * parseFloat(getComputedStyle(document.documentElement).fontSize);
+    },
+    doesOptionHaveIcon(option) {
+      // whether the given option has an icon
+      return option != undefined && option.icon != null;
+    },
+    hideDropdown() {
```

Comment:
Some lines are not covered with tests. Will you write it or skip it for now?

Reply:
need to simulate clickaway for this line which I am not sure how to do!